.github/workflows/lint.yaml

```diff
@@ -256,3 +256,38 @@ jobs:
         uses: codespell-project/codespell-problem-matcher@v1
       - name: codespell
         run: ./scripts/spelling.sh codespell
+  weblint:
+    name: weblint
+    runs-on: ubuntu-latest
+    container: debian:stable
+    steps:
+      - name: Update package database
+        run: apt-get update -y
+      - name: Install Git
+        run: apt-get -y install git
+      - uses: actions/checkout@v3
+      - name: Cache NPM
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-npm
+        with:
+          # This is the correct directory to cache
+          # See https://docs.npmjs.com/cli/v8/commands/npm-ci
+          path: ~/.npm
+          key:  debian_stable-lint-weblint-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            debian_stable-lint-weblint-${{ env.cache-name }}-
+            debian_stable-lint-weblint-
```

Comment:
Is this a match to search for, and then it finds the matching hash? Why the variants with and without the cache-name?

Reply:
Fixed by switching to `setup-node`.

Using caches with fallback priorities is recommended here: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows. I guess you can have multiple levels of caching (more specific or less specific) or caching in multiple steps for different things and try all of them if there isn't an immediate match.